### PR TITLE
`Development`: Add hint about the authorization object in the Java migration tool documentation

### PR DESCRIPTION
**Pull request by ole-ve**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
I encountered an issue when using a custom SQL query inside of a changeset with the Java Migration Tool from @julian-christl.

### Description
There seems to be a problem with the tool connecting to the database when using some custom SQL queries from a Spring repository. I encountered the following error:
`java.lang.IllegalArgumentException: Authentication object cannot be null`

### CODE REVIEW

File: docs/dev/migration/data.rst [hunk 1]
```diff
@@ -18,3 +18,4 @@ The changelog can be found in :code:`src/main/java/config/migration/MigrationReg
 - All executed entries are saved in the table :code:`migration_changelog`. If you delete entries, they get executed again.
 - Test your changes locally first, and only commit changes you are confident that work.
 - Before deploying any database changes to a test server, ask for official permission from the project lead. If the changes don’t get approved, manual rollbacks can be necessary, which are avoidable.
+- It is suggested to not use repository methods that execute custom SQL queries as this may lead to problems with the authentication to the database.
```
Comment (julian-christl) [inline docs/dev/migration/data.rst:21]:
```suggestion
- Do not use repository methods that execute custom SQL/JPQL queries as this leads to problems.
```

I think we can keep this short.

**Review by julian-christl — APPROVED** [review]

**Comment by krusche** [comment]
There is a simple solution to this problem. Simply invoke SecurityUtils.setAuthorizationObject(); if there is no user context (e.g. based on a REST Call) before

**Review by krusche — CHANGES_REQUESTED** [review]
such queries are possible, please change the docs accordingly

File: docs/dev/migration/data.rst [hunk 2]
```diff
@@ -18,3 +18,4 @@ The changelog can be found in :code:`src/main/java/config/migration/MigrationReg
 - All executed entries are saved in the table :code:`migration_changelog`. If you delete entries, they get executed again.
 - Test your changes locally first, and only commit changes you are confident that work.
 - Before deploying any database changes to a test server, ask for official permission from the project lead. If the changes don’t get approved, manual rollbacks can be necessary, which are avoidable.
+- Do not use repository methods that execute custom SQL/JPQL queries as this leads to problems.
```
Comment (julian-christl) [inline docs/dev/migration/data.rst:21]:
```suggestion
- If queries fail due to the authorization object being null, call `SecurityUtils.setAuthorizationObject()` beforehand to set a dummy object.
```

**Review by krusche — APPROVED** [review]

**Pull request merged**